packages/yourloops/pages/patient/caregivers/page.tsx

```diff
@@ -92,90 +90,73 @@ function PatientCaregiversPage(): JSX.Element {
     }
   }
 
-  const handleRemoveCaregiver = async (us: ShareUser): Promise<void> => {
-    const getConsent = async (): Promise<boolean> => {
-      return await new Promise((resolve: (consent: boolean) => void) => {
-        setCaregiverToRemove({ caregiver: us, onDialogResult: resolve })
-      })
-    }
-
-    const consent = await getConsent()
-    setCaregiverToRemove(null) // Close the dialog
-
-    if (consent && user) {
-      try {
-        if (us.status === UserInvitationStatus.pending && typeof us.invitation === 'object') {
-          await notificationHook.cancel(us.invitation)
-        } else {
-          await DirectShareApi.removeDirectShare(user.id, us.user.userid)
-        }
-        alert.success(t('modal-patient-remove-caregiver-success'))
-        setCaregivers(null) // Refresh the list
-      } catch (reason) {
-        log.error(reason)
-        alert.error(t('modal-patient-remove-caregiver-failure'))
+  const getCaregiversFromPendingInvitations = useCallback((): ShareUser[] => {
+    return sentInvitations.reduce((acc: ShareUser[], invitation: INotification) => {
+      if (invitation.type !== NotificationType.directInvitation) {
+        return acc
       }
-    }
-  }
 
-  React.useEffect(() => {
-    if (!caregivers && user && haveNotifications) {
-      // Load caregivers
-      const addPendingInvitation = (target: ShareUser[]): void => {
-        for (const invitation of sentInvitations) {
-          if (invitation.type === NotificationType.directInvitation) {
-            // Create a fake share user
-            log.debug('Found pending direct-share invitation: ', invitation)
-            const shareUser: ShareUser = {
-              invitation,
-              status: UserInvitationStatus.pending,
-              user: {
-                username: invitation.email,
-                userid: uuidv4(),
-                role: UserRoles.caregiver
-              }
-            }
-            target.push(shareUser)
-          }
+      log.debug('Found pending direct-share invitation: ', invitation)
+      const caregiver: ShareUser = {
+        invitation,
+        status: UserInvitationStatus.pending,
+        user: {
+          username: invitation.email,
+          userid: uuidv4(),
+          role: UserRoles.caregiver
         }
       }
-
-      DirectShareApi.getDirectShares().then((value) => {
-        addPendingInvitation(value)
-        setCaregivers(value)
-      }).catch((reason: unknown) => {
+      acc.push(caregiver)
+      return acc
+    }, [])
+  }, [sentInvitations])
+
+  const fetchCaregivers = useCallback(async (): Promise<void> => {
+    return await DirectShareApi.getDirectShares()
```

Comment:
The linter disagrees because of this ESlint rule: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/return-await.md